### PR TITLE
System design 21912 database design

### PR DESCRIPTION
**Pull request by le-thang-mulodo**
- # TODO LIST SYSTEM DESIGN 
  Database design:
- [x] ERD.
- [x] Table design.
- [x]  SQL script.
- [x] Local testing

### CODE REVIEW

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+-- --------------------------------------------------------
+
+--
+-- Table structure for table `category`
+--
+
+CREATE TABLE IF NOT EXISTS `category` (
+`id` int(3) NOT NULL,
+  `name` varchar(45) COLLATE utf8_unicode_ci NOT NULL,
+  `child_of_category` int(3) DEFAULT NULL,
```
Comment (ghost) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: It should be parent_id

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+-- --------------------------------------------------------
+
+--
+-- Table structure for table `comment`
+--
+
+CREATE TABLE IF NOT EXISTS `comment` (
+`id` int(11) NOT NULL,
+  `content` tinytext COLLATE utf8_unicode_ci NOT NULL,
+  `id_user` int(8) NOT NULL,
```
Comment (ghost) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: It should be: user_id, post_id

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  `id_post` int(8) NOT NULL,
+  `is_replay_to_id` int(11) NOT NULL,
```
Comment (ghost) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: What is meaning for is_replay_to_id???

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+-- --------------------------------------------------------
+
+--
+-- Table structure for table `post`
+--
+
+CREATE TABLE IF NOT EXISTS `post` (
+`id` int(8) NOT NULL,
+  `title` varchar(100) COLLATE utf8_unicode_ci NOT NULL,
```
Comment (ghost) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: Do you think, title with 100 is too short ?

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  `content` text COLLATE utf8_unicode_ci NOT NULL,
+  `image` varchar(100) COLLATE utf8_unicode_ci NOT NULL,
+  `description` text COLLATE utf8_unicode_ci NOT NULL,
+  `id_author` int(8) NOT NULL,
```
Comment (ghost) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: Should using author_id or user_id

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+-- --------------------------------------------------------
+
+--
+-- Table structure for table `post_to_catelogry`
+--
+
+CREATE TABLE IF NOT EXISTS `post_to_catelogry` (
+`id` int(4) NOT NULL,
+  `id_category` int(3) NOT NULL,
```
Comment (ghost) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: Should use category_id, post_id

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+-- --------------------------------------------------------
+
+--
+-- Table structure for table `user`
+--
+
+CREATE TABLE IF NOT EXISTS `user` (
+`id` int(8) NOT NULL,
+  `username` varchar(40) COLLATE utf8_unicode_ci NOT NULL,
```
Comment (ghost) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: First name, last name, display name should be more than 50 characters.

Comment (le-thang-mulodo) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: I changed first name, last name and display name more than 50 characters

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql (hunk 6 above)
Comment (le-thang-mulodo) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: I have changed and just keep user_id, category_id.

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql (hunk 4 above)
Comment (le-thang-mulodo) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: I have repaired

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql (hunk 5 above)
Comment (le-thang-mulodo) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: I think the user is not necessarily the author, I was changed to author_id to user_id.

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql (hunk 1 above)
Comment (le-thang-mulodo) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: I have repaired

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql (hunk 3 above)
Comment (le-thang-mulodo) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.0_ANH_THANG_LE.sql]: Under each comment may be others add more comment.  I was changed to  is_replay_to_id to parent_id.

File: doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.1_ANH_THANG_LE.sql [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+-- --------------------------------------------------------
+
+--
+-- Table structure for table `category`
+--
+
+CREATE TABLE IF NOT EXISTS `category` (
+`id` int(3) NOT NULL,
+  `name` varchar(45) COLLATE utf8_unicode_ci NOT NULL,
+  `id_parent` int(3) DEFAULT NULL,
```
Comment (ghost) [inline doc/backend/db/TRN-MiniBlog_DatabaseInstall_Ver1.1_ANH_THANG_LE.sql]: You should name for FK field like "xxxx_id", how do you think??

**Pull request merged**